Service/UserApi/UserApiService.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EMS\ClientHelperBundle\Service\UserApi;
+
+use GuzzleHttp\Client;
+
+abstract class UserApiService
+{
+    /** @var string */
+    private $baseUrl;
+
+    public function __construct(string $baseUrl)
+    {
+        $this->baseUrl = $baseUrl;
+    }
+
+    /**
+     * @param array<string, string|null> $headers
+     * @return Client
+     */
+    protected function createClient(array $headers = []): Client
+    {
+        return new Client([
+            'base_uri' => $this->baseUrl,
+            'headers'  => $headers,
+            'timeout'  => 30,
+            'allow_redirects' => false,
+        ]);
+    }
```

Comment:
```suggestion
    protected function createClient(Header $header]): Client
    {
        return new Client([
            'base_uri' => $this->baseUrl,
            'headers'  => $header->toArray(),
            'timeout'  => 30,
            'allow_redirects' => false,
        ]);
    }
```

Reply:
```suggestion
   $header = (new Header())->setContentType('application/json')->setAuthToken('letoken');
```

Reply:
I've replaced the abstract parent to client factory but I've no time now to create the Header Object. I've created an issue and I'll do that monday (really) : https://github.com/ems-project/EMSClientHelperBundle/issues/158